org.eclipse.xtext/src/org/eclipse/xtext/resource/impl/DefaultResourceDescriptionManager.java

```diff
@@ -141,15 +145,30 @@ public boolean isAffected(Collection<Delta> deltas, IResourceDescription candida
 					for(int i = 0; i < containers.size() && !descriptionIsContained; i++) {
 						descriptionIsContained = containers.get(i).hasResourceDescription(uri);
 					}
-					if (!descriptionIsContained)
-						return false;
+					if (!descriptionIsContained) {
+						return isAffectedDueToExtension(deltas, candidate, context);
+					}
 				}
 				if (isAffected(importedNames, delta.getNew()) || isAffected(importedNames, delta.getOld())) {
 					return true;
 				}
 			}
         }
-        return false;
+        return isAffectedDueToExtension(deltas, candidate, context);
+    }
+    
+    /**
+     * Query all registered extensions.
+     * 
+     * @since 2.22
+     */
+    protected boolean isAffectedDueToExtension(Collection<Delta> deltas, IResourceDescription candidate, IResourceDescriptions context) {
+    	for(int i = 0; i < isAffectedExtensions.size(); i++) {
```

Comment:
Are you using here indexed based for-loop for performance reason? Wouldn't enhanced-for perform the same?
Maybe using ```return isAffectedExtensions.stream().anyMatch(ext -> isAffected(deltas, candidate, context));` ? (May perform worse?)

The added lines have mixed space/tab indentation. Could you fix that, please?

Reply:
It's called from a tight loop from the builder. I'll suffer a few more chars here to make the JIT's life easier.